tools/bqpipeline/bqpipeline/bqpipeline.py

```diff
@@ -0,0 +1,241 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+# Copyright 2019 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#         http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+import codecs
+from google.cloud import bigquery
+from jinja2 import Template
+from jinja2.sandbox import SandboxedEnvironment
+import logging
+import sys
+
+
+def get_logger(name, fmt):
+    handler = logging.StreamHandler(sys.stdout)
+    handler.setLevel(logging.DEBUG)
+    handler.setFormatter(logging.Formatter(fmt))
+    log = logging.getLogger(name)
+    log.setLevel(logging.DEBUG)
+    log.addHandler(handler)
+    return log
+
+
+FORMAT = '%(asctime)-15s %(levelname)s %(message)s'
+QFORMAT = FORMAT + ' %(path)s\n```\n%(query)s\n```'
+logger = get_logger('job', FORMAT)
+qlog = get_logger('query', QFORMAT)
+
+
+def read_sql(path):
+    with codecs.open(path, mode='r', encoding='utf-8', buffering=-1) as f:
+        return f.read()
+
+
+def ts(project, dataset, table):
```

Comment:
it's TableSpec

Reply:
I left this as ts because it's meant be short for use in the user's pipeline code, even less characters than `'{}.{}.{}'.format(project,dataset,table)` or `'.'.join([project,dataset,table])`.